kpops/components/streams_bootstrap/producer/producer_app.py

```diff
@@ -28,6 +28,19 @@ def helm_chart(self) -> str:
             f"{self.repo_config.repository_name}/{AppType.CLEANUP_PRODUCER_APP.value}"
         )
 
+    @override
+    async def clean(self, dry_run: bool) -> None:
+        await self.update_cleaner_with_cluster_values()
+        self.app.streams.delete_output = True
+        await super().clean(dry_run)
+
+    async def update_cleaner_with_cluster_values(self):
+        """Update cleaner with cluster values if the release exists."""
+        cluster_values = await self.fetch_values_from_cluster()
+        if cluster_values:
+            self.app = ProducerAppValues.model_validate(cluster_values)
+            self.app.name_override = self.helm_name_override
+
```

Comment:
@disrupted I am using this method on the `producer_app.py` & `streams_app.py`. The difference is only `ProducerAppValues` & `StreamsAppValues`. Is there a nice Pydantic way to abstract this into the KafkaAppCleaner?

Reply:
you could try `self.app.__class__.model_validate`